packages/runtime/src/runtime/modal.tsx

```diff
@@ -0,0 +1,113 @@
+import { Modal } from "antd";
+import type { NavigateModalScreenAction } from "framework";
+import { createContext, useState } from "react";
+import { createPortal } from "react-dom";
+import { Outlet } from "react-router-dom";
+import OpenInFullIcon from "@mui/icons-material/OpenInFull";
+import CloseFullscreenIcon from "@mui/icons-material/CloseFullscreen";
+
+type ModalProps = {
+  title?: string | React.ReactNode;
+} & Omit<Exclude<NavigateModalScreenAction, string>, "name">;
+
+interface ModalContextProps {
+  visible: boolean;
+  setVisible: (visible: boolean) => void;
+  content: React.ReactNode;
+  setContent: (content: React.ReactNode) => void;
+  options: ModalProps;
+  setOptions: (options: ModalProps) => void;
+}
+
+export const ModalContext = createContext<ModalContextProps>({
+  visible: false,
+  setVisible: () => {},
```

Comment:
Can we consolidate this API? You can expose one function that calls the hooks underneath the hood to abstract away complexity